### DISCUSSION TITLE
Spacy Relation Component Unable to predict

### DISCUSSION THREAD

**Discussion by shahidkhan021** [explosion/spaCy#8627] [Help: Other Questions] [answered]
i have trained a transformer model with relation component added to pipe line by cloning 
```python -m spacy project clone tutorials/rel_component```
 then i was able to train the model successfully then i used following code for prediction 
```import random
import typer
from pathlib import Path
import spacy 
from spacy.tokens import DocBin, Doc
from spacy.training.example import Example 
from scripts.rel_pipe import make_relation_extractor, score_relations
from scripts.rel_model import create_relation_model, create_classification_layer, create_instances, create_tensors

nlp2 = spacy.load("training/model-best")
doc = nlp2(Text[0])
for name, proc in nlp2.pipeline:
  doc = proc(doc)
  for value, rel_dict in doc._.rel.items():
    for sent in doc.sents:
      for e in sent.ents:
        for b in sent.ents:
          if e.start == value[0] and b.start == value[1]:
            if rel_dict['EXPERIENCE_IN'] >=0.9 :
              print(f" entities: {e.text, b.text} --> predicted relation: {rel_dict}")
```

but when i execute above code i am getting this message
Could not determine any instances in doc - returning doc as is. can anybody help on this

**Comment by svlandeg** [comment, marked as answer]
Hi! This typically means that no relevant entities were found to create REL candidate instances with. Does your pipeline have an `ner` component?

**Reply by shahidkhan021** [reply]
when i train i am getting this message 
```ℹ Pipeline: ['transformer', 'relation_extractor']```
so i believe there is no ner pipeline sorry for this dumb question and can help me to add ner pipeline to this

**Reply by svlandeg** [reply]
Have a look at a related discussion thread here: https://github.com/explosion/spaCy/discussions/8499 ;-)
